src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -2371,6 +2425,27 @@ baseMipLevelSize: [${baseMipLevelSize.join(', ')}]
 physicalMipCount: ${physicalMipLevelCount}
   `;
 }
+
+function getComponentsToCheck(
+  device: GPUDevice,
+  format: GPUTextureFormat,
+  builtin: TextureBuiltin,
+  textureType: string
+) {
+  const returnsOneComponent = !isBuiltinGather(builtin) && isSingleChannelInput(textureType);
+  if (returnsOneComponent) {
+    return kRComponent;
+  }
+
+  const gbaUndefined =
+    isDepthOrStencilTextureFormat(format) && !device.features.has('texel-component-swizzle');
```

Comment:
@greggman The original typo still exits
```suggestion
    isDepthOrStencilTextureFormat(format) && !device.features.has('texture-component-swizzle');
```